src/sinks/datadog_metrics.rs

```diff
@@ -104,96 +109,78 @@ inventory::submit! {
-            },
-            inner: http_service,
-        };
+        Ok((Box::new(sink), healthcheck))
+    }
 
-        let sink = request
-            .batch_sink(HttpRetryLogic, datadog_http_service, cx.acker())
-            .batched_with_min(MetricBuffer::new(), &batch);
+    fn input_type(&self) -> DataType {
+        DataType::Metric
+    }
 
-        Ok(Box::new(sink))
+    fn sink_type(&self) -> &'static str {
+        "datadog_metrics"
     }
+}
 
-    fn healthcheck(config: DatadogConfig) -> crate::Result<super::Healthcheck> {
-        let uri = format!("{}/api/v1/validate?api_key={}", config.host, config.api_key)
-            .parse::<Uri>()
-            .context(super::UriParseError)?;
+impl HttpSink for DatadogSink {
+    type Input = Event;
+    type Output = Vec<Metric>;
 
-        let request = hyper::Request::get(uri).body(hyper::Body::empty()).unwrap();
+    fn encode_event(&self, event: Event) -> Option<Self::Input> {
+        Some(event)
+    }
 
-        let https = HttpsConnector::new(4).expect("TLS initialization failed");
-        let client = hyper::Client::builder().build(https);
+    fn build_request(&self, events: Self::Output) -> http::Request<Vec<u8>> {
+        let now = Utc::now().timestamp();
+        let interval = now - self.last_sent_timestamp.load(Acquire);
+        self.last_sent_timestamp.store(now, Release);
 
-        let healthcheck = client
-            .request(request)
-            .map_err(|err| err.into())
-            .and_then(|response| match response.status() {
-                StatusCode::OK => Ok(()),
-                other => Err(super::HealthcheckError::UnexpectedStatus { status: other }.into()),
-            });
+        let input = encode_events(events, interval, &self.config.namespace);
+        let body = serde_json::to_vec(&input).unwrap();
 
-        Ok(Box::new(healthcheck))
+        http::Request::get(self.uri.clone())
+            .header("Content-Type", "application/json")
+            .body(body)
+            .unwrap()
     }
 }
 
-impl Service<Vec<Metric>> for DatadogSvc {
-    type Response = HttpResponse;
-    type Error = HttpError;
-    type Future = Box<dyn Future<Item = Self::Response, Error = Self::Error> + Send + 'static>;
+fn healthcheck(config: DatadogConfig) -> crate::Result<super::Healthcheck> {
+    let uri = format!("{}/api/v1/validate?api_key={}", config.host, config.api_key)
+        .parse::<Uri>()
+        .context(super::UriParseError)?;
 
-    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
-        self.inner.poll_ready()
-    }
+    let request = hyper::Request::get(uri).body(hyper::Body::empty()).unwrap();
 
-    fn call(&mut self, items: Vec<Metric>) -> Self::Future {
-        let now = Utc::now().timestamp();
-        let interval = now - self.state.last_sent_timestamp;
-        self.state.last_sent_timestamp = now;
+    let https = HttpsConnector::new(4).expect("TLS initialization failed");
```

Comment:
I keep wondering, do we really need 4 threads for a health check?

Reply:
This should be `HttpClient` anyways.